mobile-widgets/qml/DiveList.qml

```diff
@@ -389,25 +401,18 @@ Kirigami.ScrollablePage {
 			 credentialStatus === CloudStatus.CS_VERIFIED) ? 0 : 1
 		visible: opacity > 0
 		Behavior on opacity { NumberAnimation { duration: Kirigami.Units.shortDuration } }
-		function setupActions() {
-			if (prefs.credentialStatus === CloudStatus.CS_VERIFIED ||
-					prefs.credentialStatus === CloudStatus.CS_NOCLOUD) {
-				page.actions.main = page.downloadFromDCAction
-				page.actions.right = page.addDiveAction
-				page.actions.left = page.filterToggleAction
-				page.title = qsTr("Dive list")
-				if (diveListView.count === 0)
-					showPassiveNotification(qsTr("Please tap the '+' button to add a dive (or download dives from a supported dive computer)"), 3000)
-			} else {
+		onVisibleChanged: {
+			console.log("---> Startpage changed visibility to " + visible)
+			if (visible) {
 				page.actions.main = null
 				page.actions.right = null
 				page.actions.left = null
 				page.title = qsTr("Cloud credentials")
+			} else {
+				// This is to be removed when StartPage becomes a proper page
+				page.simulateVisibleChanged()
```

Comment:
I've read this five times and it's still not clear to me if this is functionally equivalent or not.
Which is mostly a sign that the existing code (mostly done by me) is bat-shit-crazy.
So I'll trust you that you've thought this through and will test the end result :-)